### ISSUE TITLE
TESS target pixel file method to_lightcurve() unexpectedly censors data

### ISSUE THREAD

**Issue by KenMighell** [KeplerGO/lightkurve#391]
#### Problem description

TESS target pixel file method to_lightcurve() unexpectedly censors data.
With Kepler/K2 target pixel files (opened with quality_bitmask=0) the "time"
vector in the tpf file is the same size (as expected) of the "time" vector in 
the light curve produced by the to_lightcurve() method.  Unfortunately, 
this is not true with TESS target pixel files.  The to_lightcurve() may be
removing nan time values...

#### Example

```python
import lightkurve as lk

def tpf_to_lc_size_check(filename):
    print filename,'=filename'
    objf = lk.open(filename,quality_bitmask=0) 
    print type(objf),'=type(objf)  [should be a target pixel file]'
    print 
    objf.hdu.info()
    print objf.hdu[1].data['time'].size,' : size of "time" vector in the target pixel file'
    print
    lc = objf.to_lightcurve()
    print type(lc),'=type(lc) --> should be a light curve'
    print lc.time.size,' : size of "time" vector in light curve'
    print lc.flux.size,' : size of "flux" vector in light curve'
    assert(lc.flux.size == objf.hdu[1].data['time'].size),\
    '%d != %d  : the "time" vectors should be the same size!' \
    % (lc.flux.size,objf.hdu[1].data['time'].size)
    print '\nOK!  :-)'
```

```python
print '\n========== works correctly with a K2 long-cadence target pixel file ==========\n'
fn = 'ktwo212803289-c17_lpd-targ.fits'
tpf_to_lc_size_check(filename=fn)
```
RESULT:

```python
========== works correctly with a K2 long-cadence target pixel file ==========

ktwo212803289-c17_lpd-targ.fits =filename
<class 'lightkurve.targetpixelfile.KeplerTargetPixelFile'> =type(objf)  [should be a target pixel file]

Filename: ktwo212803289-c17_lpd-targ.fits
No.    Name      Ver    Type      Cards   Dimensions   Format
  0  PRIMARY       1 PrimaryHDU      54   ()      
  1  TARGETTABLES    1 BinTableHDU    287   3283R x 13C   ['D', 'E', 'J', '169J', '169E', '169E', '169E', '169E', '169E', 'J', 'E', 'E', '65E']   
  2  APERTURE      1 ImageHDU        48   (13, 13)   int32   
3283  : size of "time" vector in the target pixel file

<class 'lightkurve.lightcurve.KeplerLightCurve'> =type(lc) --> should be a light curve
3283  : size of "time" vector in light curve
3283  : size of "flux" vector in light curve

OK!  :-)
```

Now using a TESS target pixel file...

```python
print '\n========== fails using a TESS target pixel file ==========\n'
fn = 'hlsp_tess-data-alerts_tess_phot_00025155310-s01_tess_v1_tp.fits'
tpf_to_lc_size_check(filename=fn)
```
RESULT:

```python

========== fails using a TESS target pixel file ==========

hlsp_tess-data-alerts_tess_phot_00025155310-s01_tess_v1_tp.fits =filename
<class 'lightkurve.targetpixelfile.TessTargetPixelFile'> =type(objf)  [should be a target pixel file]

Filename: hlsp_tess-data-alerts_tess_phot_00025155310-s01_tess_v1_tp.fits
No.    Name      Ver    Type      Cards   Dimensions   Format
  0  PRIMARY       1 PrimaryHDU      44   ()      
  1  PIXELS        1 BinTableHDU    248   20076R x 11C   ['D', 'E', 'J', '121J', '121E', '121E', '121E', '121E', 'J', 'E', 'E']   
  2  APERTURE      1 ImageHDU        49   (11, 11)   int32   
  3  TARGET COSMIC RAY    1 BinTableHDU     27   0R x 4C   [J, I, I, E]   
20076  : size of "time" vector in the target pixel file

<class 'lightkurve.lightcurve.TessLightCurve'> =type(lc) --> should be a light curve
19261  : size of "time" vector in light curve
19261  : size of "flux" vector in light curve

---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-83-d1a17c560137> in <module>()
      4 print '\n========== fails using a TESS target pixel file ==========\n'
      5 fn = 'hlsp_tess-data-alerts_tess_phot_00025155310-s01_tess_v1_tp.fits'
----> 6 tpf_to_lc_size_check(filename=fn)
      7 print '\n==========\n'

<ipython-input-82-6faa40fb5033> in tpf_to_lc_size_check(filename)
     12     print lc.time.size,' : size of "time" vector in light curve'
     13     print lc.flux.size,' : size of "flux" vector in light curve'
---> 14     assert(lc.flux.size == objf.hdu[1].data['time'].size),    '%d != %d  : the "time" vectors should be the same size!' % (lc.flux.size,objf.hdu[1].data['time'].size)
     15     print '\nOK!  :-)'

AssertionError: 19261 != 20076  : the "time" vectors should be the same size!
```

#### Expected behavior

The flux and time vectors produced by the to_lightcurve method should be the same size
as the time vector in the tpf file.    I have come up with a hack to fix this but it does make
code handling target pixel files more complicated with a version for Kepler/K2 and a version
for TESS.

#### Environment:
- platform (e.g. Linux, OSX, Windows): Mac OS X El Capitan (10.11.5)
- lightkurve version: 1.0b26.dev
- installation method: github source

**Comment by KenMighell** [comment]
```python
class TessTargetPixelFile(TargetPixelFile):
    """
    Defines a TargetPixelFile class for the TESS Mission.
    Enables extraction of raw lightcurves and centroid positions.

    Parameters
    ----------
    path : str
        Path to a Kepler Target Pixel (FITS) File.
    quality_bitmask : str or int
        Bitmask specifying quality flags of cadences that should be ignored.
    kwargs : dict
        Keyword arguments passed to `astropy.io.fits.open()`.
    """
    def __init__(self, path, quality_bitmask='default', **kwargs):
        super(TessTargetPixelFile, self).__init__(path,
                                                  quality_bitmask=quality_bitmask,
                                                  **kwargs)
        self.quality_mask = TessQualityFlags.create_quality_mask(
                                quality_array=self.hdu[1].data['QUALITY'],
                                bitmask=quality_bitmask)
        # Early TESS releases had cadences with time=NaN (i.e. missing data)
        # which were not flagged by a QUALITY flag yet; the line below prevents
        # these cadences from being used. They would break most methods!
        self.quality_mask &= np.isfinite(self.hdu[1].data['TIME'])
```

**Comment by barentsen** [comment]
@KenMighell I would not object to see the `if (quality_bitmask > 0):` statement added!  Would you be willing to open a PR to add this?  (We have a guide on how to open a PR here: https://docs.lightkurve.org/contributing.html#proposing-changes-to-lightkurve-using-github-pull-requests )

I'm going to be on leave until Friday, but happy to have me or someone else merge a PR that implements this change!  Thanks.

**Comment by barentsen** [comment]
I will also add that it is my understanding that future TESS data releases will flag the offending cadences that have `time=nan`, at which point we should definitely remove the hack that filters them out!

**Comment by barentsen** [comment]
This was resolved by #392. :tada:
